CT-1542 Implement Business Unit performance report for SARs

stephenrichards opened the pull request:
This is implemented by moving the logic from the existing
R003BusinessUnitPeformanceReport into a new class BaseBusinessUnitPerformanceRepoert
and deriving the Business unit performce reports for both FOIs and SARs from that.  The
derived classes just differ in returning different names nad desciptions of the report, and
most importantly the case_scope, which determines which cases will be reported on

The specs for the new R103SarBusinessUnitPerformanceReport only exercise the new methods; the
code in the base class for analysing, totalling, calculating percentages in thoroughly testing in
the spec for the FOI report.

misaka commented on db/migrate/20170713094438_add_request_clarification_ooutcome.rb:
```diff
@@ -0,0 +1,9 @@
+class AddRequestClarificationOoutcome < ActiveRecord::Migration[5.0]
```
Is this a distant relative to the Oort Cloud? ;)

stephenrichards replied:
yep
:-)

stephenrichards replied:
It's only the name of a  migration, so let's leave it for now and get it in.

stephenrichards committed:
CT-1542 Implement Business Unit performance report for SARs

This is implemented by moving the logic from the existing
R003BusinessUnitPeformanceReport into a new class BaseBusinessUnitPerformanceRepoert
and deriving the Business unit performce reports for both FOIs and SARs from that.  The
derived classes just differ in returning different names nad desciptions of the report, and
most importantly the case_scope, which determines which cases will be reported on

The specs for the new R103SarBusinessUnitPerformanceReport only exercise the new methods; the
code in the base class for analysing, totalling, calculating percentages in thoroughly testing in
the spec for the FOI report.

stephenrichards committed:
Merge branch 'master' into ct-1542-sar-reports
